hazelcast/src/main/java/com/hazelcast/sql/impl/JetSqlSerializerHook.java

```diff
@@ -50,12 +49,22 @@ public int getFactoryId() {
     @SuppressWarnings("unchecked")
     @Override
     public DataSerializableFactory createFactory() {
+
         ConstructorFunction<Integer, IdentifiedDataSerializable>[] constructors = new ConstructorFunction[LEN];
 
-        constructors[JSON_QUERY] = arg -> new JsonQueryFunction();
-        constructors[JSON_PARSE] = arg -> new JsonParseFunction();
-        constructors[JSON_VALUE] = arg -> new JsonValueFunction();
+        constructors[JSON_QUERY] = arg -> SERIALIZATION_FACTORY.createJsonQueryFunction();
+        constructors[JSON_PARSE] = arg -> SERIALIZATION_FACTORY.createJsonParseFunction();
+        constructors[JSON_VALUE] = arg -> SERIALIZATION_FACTORY.createJsonValueFunction();
 
         return new ArrayDataSerializableFactory(constructors);
     }
+
+    private static SqlSerializationFactory lookupSerializationFactory() {
+        try {
+            final Class<?> clazz = Class.forName("com.hazelcast.jet.sql.impl.JetSqlSerializationFactory");
+            return (SqlSerializationFactory) clazz.newInstance();
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException ignored) {
+            return new NoopSqlSerializationFactory();
```

Comment:
- I'd move this class as an inner static class, it's strictly local and sufficiently small
- Instead of returning we should rather throw something like `hazelcast-sql module not available on class path`.
- I'd also rename it from `Noop...` to `Failing...`

Reply:
The `DataSerializableFactory.create` factory methods actually allows null return value, but that's for the case when the ID isn't know. But in this case the ID is know, but we failed to create instance.